java-lib/src/main/java/org/clink/feature/onehotencoder/ClinkOneHotEncoderModel.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.ml.common.datastream.TableUtils;
+import org.apache.flink.ml.common.param.HasHandleInvalid;
+import org.apache.flink.ml.feature.onehotencoder.OneHotEncoderModel;
+import org.apache.flink.ml.linalg.SparseVector;
+import org.apache.flink.ml.util.ReadWriteUtils;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.Table;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.api.internal.TableImpl;
+import org.apache.flink.table.runtime.typeutils.ExternalTypeInfo;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import com.sun.jna.Pointer;
+import org.apache.commons.lang3.ArrayUtils;
+import org.clink.linalg.ClinkSparseVector;
+import org.clink.util.ClinkKernels;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.Collections;
+import java.util.Vector;
+
+/**
+ * Wrapper class for Flink ML OneHotEncoderModel which calls equivalent C++ operator to transform.
+ */
+public class ClinkOneHotEncoderModel extends OneHotEncoderModel {
+    private String modelPath;
+
+    @Override
+    public Table[] transform(Table... inputs) {
+        final String[] inputCols = getInputCols();
+        final String[] outputCols = getOutputCols();
+
+        Preconditions.checkArgument(getHandleInvalid().equals(HasHandleInvalid.ERROR_INVALID));
+        Preconditions.checkArgument(inputs.length == 1);
+        Preconditions.checkArgument(inputCols.length == outputCols.length);
+
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldTypes(),
+                                Collections.nCopies(
+                                                outputCols.length,
+                                                ExternalTypeInfo.of(Vector.class))
+                                        .toArray(new TypeInformation[0])),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), outputCols));
+
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+
+        DataStream<Row> output =
+                tEnv.toDataStream(inputs[0])
+                        .map(
+                                new GenerateOutputsFunction(modelPath, getInputCols()),
+                                outputTypeInfo);
+
+        return new Table[] {tEnv.fromDataStream(output)};
+    }
+
+    private static class GenerateOutputsFunction extends RichMapFunction<Row, Row> {
+        private final String modelPath;
+        private Pointer modelPointer;
```

Comment:
It seems cleaner to keep `modelPointer` as a member variable of `ClinkOneHotEncoderModel` and de-allocate this pointer when the `ClinkOneHotEncoderModel` is destructed.

Reply:
As discussed offline, reason is the same as above.